uaclient/entitlements/__init__.py

```diff
@@ -217,3 +226,55 @@ def create_enable_entitlements_not_found_error(
         invalid_service=", ".join(entitlements_not_found),
         service_msg=service_msg,
     )
+
+
+# This function is just here to simplify our unittests, as
+# mocking isinstance directly doesn't work here
+def _is_repo_entitlement(ent):
+    return isinstance(ent, RepoEntitlement)
+
+
+def check_entitlement_directives_are_unique(
+    cfg: UAConfig,
+) -> Tuple[bool, Optional[EntitlementDirectiveClash]]:
```

Comment:
Could this function `raise exceptions.EntitlementsDirectivesAreNotUnique` directly?

Reply:
Yes, that's a better approach